utils/reflection/reflection_test.go

```diff
@@ -487,3 +488,92 @@ func TestIsEmpty(t *testing.T) {
 		})
 	}
 }
+
+func TestToStructPtr(t *testing.T) {
+	vInt := 15
+	vStr := faker.Sentence()
+	var vPrt *string
+	vBytes := []byte(faker.Sentence())
+	vBool := false
+	vfloat := 150.454
+	vMap := map[string]string{faker.Word(): faker.Sentence()}
+	vArray := []string{faker.Word(), faker.Sentence(), faker.Name()}
+	vStruct := struct {
+		Test  string
+		Test2 int
+		test3 *string
+	}{
+		Test:  vStr,
+		Test2: vInt,
+		test3: vPrt,
+	}
+	tests := []struct {
+		input          interface{}
+		expectedOutput interface{}
+		expectedError  error
+	}{
+		{
+			input:          nil,
+			expectedOutput: nil,
+			expectedError:  commonerrors.ErrUnsupported,
```

Comment:
Are there any other cases that would cause the error or is nil the only one? Maybe something that will cause `!vp.CanInterface` to error and another that causes `!obj.IsValid` to error?

If you pass it a `var v interface{}` it should return an error shouldn't it?

Reply:
I added more test cases for ensuring both conditions are tested.
From what I could see, canInterface only return false for not exposed structure fields https://stackoverflow.com/questions/50279840/when-is-go-reflect-caninterface-false